### PR TITLE
fix: remove OIDC auth from Http Api

### PR DESCRIPTION
**Pull request by keetonian**
*Issue #, if available:*
N/A

*Description of changes:*
OIDC and OAuth2 are the same authorization on the backend and the configuration for both is practically the same. This was causing some confusion for customers (and for us), so we decided to remove OIDC and only support the OAuth2 strategy for GA for Http Apis.

*Description of how you validated changes:*
Local and cloud testing

*Checklist:*

- [x] Write/update tests
- [x] `make pr` passes
- [x] Update documentation
- [ ] Verify transformed template deploys and application functions as expected
- [ ] Add/update example to `examples/2016-10-31`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: samtranslator/model/apigatewayv2.py [hunk 1]
```diff
@@ -57,33 +57,24 @@ class ApiGatewayV2ApiMapping(Resource):
...
     ):
         """
         Creates an authorizer for use in V2 Http Apis
         """
-        # OIDC uses a connect url, oauth2 doesn't
-        self.auth_type = "openIdConnect"
-        if open_id_connect_url is None:
-            self.auth_type = "oauth2"
+        # Currently only one type of auth
+        self.auth_type = "oauth2"
```
Comment (ShreyaGangishetty) [inline samtranslator/model/apigatewayv2.py]: can `oauth2` take `open_id_connect_url`?

**Comment by alexw91** [comment]
# [Codecov](https://codecov.io/gh/awslabs/serverless-application-model/pull/1491?src=pr&el=h1) Report
> Merging [#1491](https://codecov.io/gh/awslabs/serverless-application-model/pull/1491?src=pr&el=desc) into [develop](https://codecov.io/gh/awslabs/serverless-application-model/commit/ef92c3f7ab2efec8d819d4786bc3d4ce48475438&el=desc) will **increase** coverage by `0.02%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/awslabs/serverless-application-model/pull/1491?src=pr&el=tree)

```diff
@@             Coverage Diff             @@
##           develop    #1491      +/-   ##
===========================================
+ Coverage    94.10%   94.12%   +0.02%     
===========================================
  Files           78       78              
  Lines         4900     4921      +21     
  Branches       988      989       +1     
===========================================
+ Hits          4611     4632      +21     
  Misses         134      134              
  Partials       155      155              
```


| [Impacted Files](https://codecov.io/gh/awslabs/serverless-application-model/pull/1491?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [samtranslator/model/sam\_resources.py](https://codecov.io/gh/awslabs/serverless-application-model/pull/1491/diff?src=pr&el=tree#diff-c2FtdHJhbnNsYXRvci9tb2RlbC9zYW1fcmVzb3VyY2VzLnB5) | `94.29% <0.00%> (ø)` | |
| [samtranslator/model/eventsources/push.py](https://codecov.io/gh/awslabs/serverless-application-model/pull/1491/diff?src=pr&el=tree#diff-c2FtdHJhbnNsYXRvci9tb2RlbC9ldmVudHNvdXJjZXMvcHVzaC5weQ==) | `90.78% <0.00%> (ø)` | |
| [samtranslator/plugins/globals/globals.py](https://codecov.io/gh/awslabs/serverless-application-model/pull/1491/diff?src=pr&el=tree#diff-c2FtdHJhbnNsYXRvci9wbHVnaW5zL2dsb2JhbHMvZ2xvYmFscy5weQ==) | `99.05% <0.00%> (ø)` | |
| [...translator/plugins/api/implicit\_rest\_api\_plugin.py](https://codecov.io/gh/awslabs/serverless-application-model/pull/1491/diff?src=pr&el=tree#diff-c2FtdHJhbnNsYXRvci9wbHVnaW5zL2FwaS9pbXBsaWNpdF9yZXN0X2FwaV9wbHVnaW4ucHk=) | `100.00% <0.00%> (ø)` | |
| [samtranslator/plugins/api/implicit\_api\_plugin.py](https://codecov.io/gh/awslabs/serverless-application-model/pull/1491/diff?src=pr&el=tree#diff-c2FtdHJhbnNsYXRvci9wbHVnaW5zL2FwaS9pbXBsaWNpdF9hcGlfcGx1Z2luLnB5) | `98.24% <0.00%> (+0.01%)` | :arrow_up: |
| [samtranslator/model/api/http\_api\_generator.py](https://codecov.io/gh/awslabs/serverless-application-model/pull/1491/diff?src=pr&el=tree#diff-c2FtdHJhbnNsYXRvci9tb2RlbC9hcGkvaHR0cF9hcGlfZ2VuZXJhdG9yLnB5) | `92.27% <0.00%> (+0.06%)` | :arrow_up: |
| [samtranslator/model/apigatewayv2.py](https://codecov.io/gh/awslabs/serverless-application-model/pull/1491/diff?src=pr&el=tree#diff-c2FtdHJhbnNsYXRvci9tb2RlbC9hcGlnYXRld2F5djIucHk=) | `93.93% <0.00%> (-0.80%)` | :arrow_down: |
| [...translator/plugins/api/implicit\_http\_api\_plugin.py](https://codecov.io/gh/awslabs/serverless-application-model/pull/1491/diff?src=pr&el=tree#diff-c2FtdHJhbnNsYXRvci9wbHVnaW5zL2FwaS9pbXBsaWNpdF9odHRwX2FwaV9wbHVnaW4ucHk=) | `97.43% <0.00%> (+0.94%)` | :arrow_up: |

------

[Continue to review full report at Codecov](https://codecov.io/gh/awslabs/serverless-application-model/pull/1491?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/awslabs/serverless-application-model/pull/1491?src=pr&el=footer). Last update [ef92c3f...4dbc15a](https://codecov.io/gh/awslabs/serverless-application-model/pull/1491?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

File: samtranslator/model/apigatewayv2.py (hunk 1 above)
Comment (keetonian) [reply samtranslator/model/apigatewayv2.py]: Nope, that's why I ripped it out everywhere. I think I got all mentions of it.

**Comment by keetonian** [comment]
Added an explicit error in latest commit.

**Comment by alexw91** [comment]
# [Codecov](https://codecov.io/gh/awslabs/serverless-application-model/pull/1491?src=pr&el=h1) Report
> Merging [#1491](https://codecov.io/gh/awslabs/serverless-application-model/pull/1491?src=pr&el=desc) into [develop](https://codecov.io/gh/awslabs/serverless-application-model/commit/ef92c3f7ab2efec8d819d4786bc3d4ce48475438&el=desc) will **increase** coverage by `0.02%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/awslabs/serverless-application-model/pull/1491?src=pr&el=tree)

```diff
@@             Coverage Diff             @@
##           develop    #1491      +/-   ##
===========================================
+ Coverage    94.10%   94.13%   +0.02%     
===========================================
  Files           78       78              
  Lines         4900     4925      +25     
  Branches       988      990       +2     
===========================================
+ Hits          4611     4636      +25     
  Misses         134      134              
  Partials       155      155              
```


| [Impacted Files](https://codecov.io/gh/awslabs/serverless-application-model/pull/1491?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [samtranslator/model/apigatewayv2.py](https://codecov.io/gh/awslabs/serverless-application-model/pull/1491/diff?src=pr&el=tree#diff-c2FtdHJhbnNsYXRvci9tb2RlbC9hcGlnYXRld2F5djIucHk=) | `93.93% <0.00%> (-0.80%)` | :arrow_down: |
| [samtranslator/model/sam\_resources.py](https://codecov.io/gh/awslabs/serverless-application-model/pull/1491/diff?src=pr&el=tree#diff-c2FtdHJhbnNsYXRvci9tb2RlbC9zYW1fcmVzb3VyY2VzLnB5) | `94.29% <0.00%> (ø)` | |
| [samtranslator/model/eventsources/push.py](https://codecov.io/gh/awslabs/serverless-application-model/pull/1491/diff?src=pr&el=tree#diff-c2FtdHJhbnNsYXRvci9tb2RlbC9ldmVudHNvdXJjZXMvcHVzaC5weQ==) | `90.78% <0.00%> (ø)` | |
| [samtranslator/plugins/globals/globals.py](https://codecov.io/gh/awslabs/serverless-application-model/pull/1491/diff?src=pr&el=tree#diff-c2FtdHJhbnNsYXRvci9wbHVnaW5zL2dsb2JhbHMvZ2xvYmFscy5weQ==) | `99.05% <0.00%> (ø)` | |
| [...translator/plugins/api/implicit\_rest\_api\_plugin.py](https://codecov.io/gh/awslabs/serverless-application-model/pull/1491/diff?src=pr&el=tree#diff-c2FtdHJhbnNsYXRvci9wbHVnaW5zL2FwaS9pbXBsaWNpdF9yZXN0X2FwaV9wbHVnaW4ucHk=) | `100.00% <0.00%> (ø)` | |
| [samtranslator/plugins/api/implicit\_api\_plugin.py](https://codecov.io/gh/awslabs/serverless-application-model/pull/1491/diff?src=pr&el=tree#diff-c2FtdHJhbnNsYXRvci9wbHVnaW5zL2FwaS9pbXBsaWNpdF9hcGlfcGx1Z2luLnB5) | `98.24% <0.00%> (+0.01%)` | :arrow_up: |
| [samtranslator/model/api/http\_api\_generator.py](https://codecov.io/gh/awslabs/serverless-application-model/pull/1491/diff?src=pr&el=tree#diff-c2FtdHJhbnNsYXRvci9tb2RlbC9hcGkvaHR0cF9hcGlfZ2VuZXJhdG9yLnB5) | `92.40% <0.00%> (+0.18%)` | :arrow_up: |
| [...translator/plugins/api/implicit\_http\_api\_plugin.py](https://codecov.io/gh/awslabs/serverless-application-model/pull/1491/diff?src=pr&el=tree#diff-c2FtdHJhbnNsYXRvci9wbHVnaW5zL2FwaS9pbXBsaWNpdF9odHRwX2FwaV9wbHVnaW4ucHk=) | `97.43% <0.00%> (+0.94%)` | :arrow_up: |

------

[Continue to review full report at Codecov](https://codecov.io/gh/awslabs/serverless-application-model/pull/1491?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/awslabs/serverless-application-model/pull/1491?src=pr&el=footer). Last update [ef92c3f...ea60b27](https://codecov.io/gh/awslabs/serverless-application-model/pull/1491?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Pull request merged**